instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -26,6 +26,87 @@
 from opentelemetry.semconv.trace import NetTransportValues, SpanAttributes
 from opentelemetry.trace.status import Status, StatusCode
 
+sql_reserved_words = [
+    "ADD",
+    "ALL",
+    "ALTER",
+    "AND",
+    "ANY",
+    "AS",
+    "ASC",
+    "BACKUP",
+    "BETWEEN",
+    "CASE",
+    "CHECK",
+    "COLUMN",
+    "CONSTRAINT",
+    "CREATE",
+    "DATABASE",
+    "DEFAULT",
+    "DELETE",
+    "DESC",
+    "DISTINCT",
+    "DROP",
+    "EXEC",
+    "EXISTS",
+    "FOREIGN",
+    "FROM",
+    "FULL",
+    "GROUP",
+    "BY",
+    "HAVING",
+    "IN",
+    "INDEX",
+    "INNER",
+    "INSERT",
+    "INTO",
+    "IS",
+    "JOIN",
+    "KEY",
+    "LEFT",
+    "LIKE",
+    "LIMIT",
+    "NOT",
+    "NULL",
+    "ON",
+    "OR",
+    "ORDER",
+    "OUTER",
+    "PRIMARY",
+    "PROCEDURE",
+    "RIGHT",
+    "ROWNUM",
+    "SELECT",
+    "SET",
+    "TABLE",
+    "TOP",
+    "TRUNCATE",
+    "UNION",
+    "UNIQUE",
+    "UPDATE",
+    "VALUES",
+    "VIEW",
+    "WHERE",
+    "=",
+]
+
+sql_reserved_dict = {word: True for word in sql_reserved_words}
+
+
+def _sanitize_query(query):
+    """Remove query content, replace with sanitization symbol.
+    For example `SELECT * FROM table` will sanitize to SELECT ? FROM ?`
+    """
+    sanitized_query = ""
+    if not query:
+        return sanitized_query
+
+    for word in query.split():
+        if word.upper() not in sql_reserved_dict:
+            word = "?"
+        sanitized_query += word + " "
```

Comment:
While this is better than the last change I still find this naive because I can think of some `WHERE` clause cases this breaks. I wonder if there is a sqlalchemy does this sanitization work more reliably?

Reply:
The idea behind this is that anything which isn't a reserved word will be sanitized. Can you clarify with an example of where clause case which breaks this?

Reply:
Imagine a scenario `WHERE` clause filter containing spaces and one of these reserved words. How does it behave for the following query

```sql
SELECT * FROM table WHERE column_name="PRIMARY BANKING STOCK INDEX COLLAPSE". 
```

Reply:
Hmm yes, I see what you're saying. This will result with `SELECT ? FROM table WHERE ? ? ? INDEX ?`. I still think the `SELECT ? ?` option is the safest